composer.json

```diff
@@ -14,20 +14,22 @@
     ],
     "require": {
         "php": "^7.1",
-        "symfony/options-resolver": "^3.3.6 || ^4.0",
-        "symfony/property-access": "^3.3.6 || ^4.0",
+        "psr/container": "^1.0.0",
         "symfony/intl": "^3.3.6 || ^4.0",
-        "psr/container": "^1.0.0"
+        "symfony/options-resolver": "^3.3.6 || ^4.0",
+        "symfony/property-access": "^3.3.6 || ^4.0"
     },
     "require-dev": {
         "api-platform/core": "^2.0.10",
         "doctrine/dbal": "^2.5.5",
         "doctrine/doctrine-bundle": "^1.7",
         "doctrine/doctrine-cache-bundle": "^1.3",
         "doctrine/orm": "^2.5",
+        "friendsofphp/php-cs-fixer": "^2.7",
```

Comment:
Unless it affects the code itself it's better not to do this as it can easily break with other constraints.
These tools modify the code but don't run it.

Reply:
Not sure what you mean here.

The idea is to be able to run the build process for yourself fully, without Travis-specific hacks.

For example, while developing, you'd do `make test`, `make test`, `make test`. When done, you'd run a `make dist` which would run the full test suite, PHPStan, CS Fixer etc.

After you push, CI runs the same thing, but in "check only" mode, so CS Fixer would do a dry-run, it wouldn't modify the code there.

Reply:
There's also https://github.com/jakzal/phpqa (or my version of it https://github.com/dkarlovi/php-toolbox), it's Docker-based, but you need Docker to run the full test suite anyway.

Reply:
I would vote for a Docker solution as it's easier to install (somewhat 😄).

The problem with requiring tools as dependencies is that there constraints can affect the 
libraries constraints.

For example https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/2.7/composer.json#L28 
this is a dependency also used by RollerworksSearch, but if they didn't support Symfony 4 we wouldn't be able to use Symfony 4 (_for testing compatibility_).
For PHPUnit it makes sense as the library uses the code and interfaces, and there have been a number of problems with using the phar in the past.

But PHP-cs-fixer and PHP-Stan only modify the code without actually executing it.
Therefor a lower/higher constraint shouldn't really affect our codebase.

Reply:
Yeah, I don't think Composer should install my dev dependencies' dev dependencies (that's a keyboard-full to write) by default.

We can go with https://github.com/jakzal/phpqa then.